apps/studio/src/lib/db/clients/mysql.ts

```diff
@@ -0,0 +1,1340 @@
+
+function getRealError(conn, err) {
+  /* eslint no-underscore-dangle:0 */
+  if (conn && conn._protocol && conn._protocol._fatalError) {
+    logger().warn("Query error", err, conn._protocol._fatalError);
+    return conn._protocol._fatalError;
+  }
+  return err;
+}
+
+function configDatabase(
+  server: IDbConnectionServer,
+  database: IDbConnectionDatabase
+): mysql.PoolOptions {
+  const config: mysql.PoolOptions = {
+    host: server.config.host,
+    port: server.config.port,
+    user: server.config.user,
+    password: server.config.password,
+    database: database.database,
+    multipleStatements: true,
+    dateStrings: true,
+    supportBigNumbers: true,
+    bigNumberStrings: true,
+    connectTimeout: 60 * 60 * 1000,
+  };
+
+  if (server.config.socketPathEnabled) {
+    config.socketPath = server.config.socketPath;
+    config.host = null;
+    config.port = null;
+    return config;
+  }
+
+  if (server.sshTunnel) {
+    config.host = server.config.localHost;
+    config.port = server.config.localPort;
+  }
+
+  if (server.config.ssl) {
+    config.ssl = {};
+
+    if (server.config.sslCaFile) {
+      /* eslint-disable-next-line */
+      // @ts-ignore
+      config.ssl.ca = readFileSync(server.config.sslCaFile);
+    }
+
+    if (server.config.sslCertFile) {
+      /* eslint-disable-next-line */
+      // @ts-ignore
+      config.ssl.cert = readFileSync(server.config.sslCertFile);
+    }
+
+    if (server.config.sslKeyFile) {
+      /* eslint-disable-next-line */
+      // @ts-ignore
+      config.ssl.key = readFileSync(server.config.sslKeyFile);
+    }
+
+    if (!config.ssl.key && !config.ssl.ca && !config.ssl.cert) {
+      // TODO: provide this as an option in settings
+      // or per-connection as 'reject self-signed certs'
+      // How it works:
+      // if false, cert can be self-signed
+      // if true, has to be from a public CA
+      // Heroku certs are self-signed.
+      // if you provide ca/cert/key files, it overrides this
+      config.ssl.rejectUnauthorized = false;
+    } else {
+      config.ssl.rejectUnauthorized = server.config.sslRejectUnauthorized;
+    }
+  }
+
+  return config;
+}
+
+function resolveDefault(
+  version: MysqlClient["versionInfo"],
+  defaultValue: string
+) {
+  // adapted from https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/998/files
+  if (version.isMySql) return defaultValue;
+
+  if (!defaultValue) return null;
+
+  if (defaultValue.toString().toLowerCase() === "'null'") {
+    return null;
+  }
+
+  if (
+    defaultValue.startsWith("'") &&
+    defaultValue.endsWith("'") &&
+    defaultValue.length >= 2
+  ) {
+    // MariaDb escapes all single quotes with two single quotes in default value strings, even if they are
+    // escaped with backslashes in the original `CREATE TABLE` statement.
+    return defaultValue
+      .substring(1, defaultValue.length - 1)
+      .replaceAll("''", "'");
+  }
+
+  return defaultValue;
```

Comment:
We don't actually need to branch here. This function could just return `defaultValue` here, and then override it in the MariaDB class to do all that extra logic